src/distribution_spatial.cpp

```diff
@@ -306,25 +306,25 @@ Position MeshSpatial::sample(uint64_t* seed) const
 
 PointCloud::PointCloud(pugi::xml_node node)
 {
-  std::vector<double> coords;
-
   if (check_for_node(node, "coords")) {
     point_cloud_ = get_node_position_array(node, "coords");
   } else {
     fatal_error("No coordinates were provided for the PointCloud "
                 "spatial distribution");
   }
 
-  std::vector<double> strengths(point_cloud_.size(), 1.0);
+  std::vector<double> strengths;
 
-  if (check_for_node(node, "strengths")) {
+  if (check_for_node(node, "strengths"))
     strengths = get_node_array<double>(node, "strengths");
-    if (strengths.size() != point_cloud_.size()) {
-      fatal_error(
-        fmt::format("Number of entries for the strengths array {} does "
-                    "not match the number of spatial points provided {}.",
-          strengths.size(), point_cloud_.size()));
-    }
+  else
+    strengths.resize(point_cloud_.size(), 1.0);
+
+  if (strengths.size() != point_cloud_.size()) {
+    fatal_error(
+      fmt::format("Number of entries for the strengths array {} does "
+                  "not match the number of spatial points provided {}.",
+        strengths.size(), point_cloud_.size()));
   }
```

Comment:
again - this was all modeled on `MeshSpatial` - I'm not sure I agree with this change - there should be no reason to test the size of `strengths` except in the case where it is trying to extract XML data.  Moving this test outside of that block seems like overkill.

Reply:
The control flow feels better to me in this format - fewer levels of indentation doesn't interrupt the logic of the statements above where `strengths` is populated. Placing such a check at the end of the constructor makes sense to me as regardless of how strengths is created above, this condition should be true. It occurs in the constructor, executed once, and shouldn't be expensive. Happy to reconsider if you feel strongly about the previous control flow.

Reply:
Thanks for being on board with the updates pushed here. I have a branch with similar changes for the `MeshSpatial` class, so a similar philosophy is adopted there. I'll create a PR with those (and any other improvements that apply there as well) once this has been merged.